### PR TITLE
WIP MueLu: Remove Epetra support

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/xpetra @trilinos/muelu 

## Motivation
If merged, this PR would do the following:
- Remove `MueLu_ENABLE_Kokkos_Refactor` and instead always build the Kokkos version of factories. (The parameter `use kokkos refactor` is not affected by this.)
- Remove `Xpetra_ENABLE_Kokkos_Refactor` and make Tpetra a hard dependency for Xpetra. (Needed, as we rely on `Tpetra::LocalMap`.)
- Remove Epetra, Amesos, Ifpack, AztecOO from MueLu. Make Tpetra, Ifpack2 and Amesos2 hard dependencies.

The first item could probably be merged soon. The remainder requires a bit more discussion. But I want to see if this builds and runs across multiple configurations of the AT. 

What's still missing / to be discussed:
- Xpetra still has the optional Epetra dependency.
- There are lots of leftovers and mentions of Epetra in MueLu.
- A few MueLu tests don't have a Tpetra version.
- Muemex

### CODE REVIEW

File: packages/muelu/matlab/src/MueLu_MatlabUtils_decl.hpp [hunk 1]
```diff
@@ -49,173 +49,5 @@
 
 #include "MueLu_ConfigDefs.hpp"
 
-#if !defined(HAVE_MUELU_MATLAB) || !defined(HAVE_MUELU_EPETRA) || !defined(HAVE_MUELU_TPETRA)
 #error "Muemex requires MATLAB, Epetra and Tpetra."
```
Comment (GrahamBenHarper) [inline packages/muelu/matlab/src/MueLu_MatlabUtils_decl.hpp:52]: Hm. Does this break MueMex?

Comment (cgcgcg) [reply packages/muelu/matlab/src/MueLu_MatlabUtils_decl.hpp:52]: Probably

**Comment by csiefer2** [comment]
Do we still have algorithms that aren't in the Kokkos factory path?

**Comment by cgcgcg** [comment]
Cut-drop has not Kokkos factory path. I'm sure there are others. I'm not removing non-Kokkos factories.

**Comment by mayrmt** [comment]
@cgcgcg What's the timeline for this effort? Our code still uses Epetra and we are in the process of migrating to Tpetra. Just want to get a feeling, if we need to hurry or still have some time to complete the migration.

**Comment by cgcgcg** [comment]
@mayrmt My guess is you still have some time. I don't think this will get merged anytime soon.

**Pull request closed without merging**